### PR TITLE
Added commit approved notification type

### PR DESCRIPTION
**Pull request by danielwaterworth**

### CODE REVIEW

**Comment by jjaffeux** [comment]
@danielwaterworth please make sure you follow https://meta.discourse.org/t/github-checkin-prefix-convention/19392

File: app/assets/javascripts/discourse/widgets/notification-item.js.es6 [hunk 1]
```diff
@@ -118,6 +119,16 @@ createWidget("notification-item", {
       return I18n.t(scope, { count, group_name });
     }
 
+    if (notificationType === COMMIT_APPROVED_TYPE) {
+      const num_approved_commits = data.num_approved_commits;
```
Comment (eviltrout) [inline app/assets/javascripts/discourse/widgets/notification-item.js.es6]: Our variables in JS should be camelCase

File: app/assets/javascripts/discourse/widgets/notification-item.js.es6 [hunk 2]
```diff
@@ -118,6 +119,16 @@ createWidget("notification-item", {
...
+
+      if (num_approved_commits == 1) {
+        return I18n.t(scope + '.one');
```
Comment (jjaffeux) [inline app/assets/javascripts/discourse/widgets/notification-item.js.es6]:
```suggestion
        return I18n.t(`${scope}.one`);
```

File: app/assets/javascripts/discourse/widgets/notification-item.js.es6 [hunk 3]
```diff
@@ -118,6 +119,16 @@ createWidget("notification-item", {
...
+      } else {
+        return I18n.t(scope + '.other', { num_approved_commits });
```
Comment (jjaffeux) [inline app/assets/javascripts/discourse/widgets/notification-item.js.es6]:
```suggestion
        return I18n.t(`${scope}.other`, { num_approved_commits });
```

File: app/models/notification.rb [hunk 4]
```diff
@@ -63,7 +63,8 @@ def self.types
                         watching_first_post: 17,
                         topic_reminder: 18,
                         liked_consolidated: 19,
-                        post_approved: 20
+                        post_approved: 20,
+                        commit_approved: 21
```
Comment (SamSaffron) [inline app/models/notification.rb]: To me this is probably the scope of the entire change. We just want the number reserved in core, the rest of the functionality / strings and so on can live in the plugin imo.

Comment (danielwaterworth) [reply app/models/notification.rb]: Ok, I'll work out how to monkey-patch the other changes in.

**Comment by danielwaterworth** [comment]
@jjaffeux, @eviltrout, Thanks for the feedback.

File: app/models/notification.rb (hunk 4 above)
Comment (SamSaffron) [reply app/models/notification.rb]: Prefer a less monkey patchy way, can you expose an API for it from the JS side?

Comment (danielwaterworth) [reply app/models/notification.rb]: Ok

**Comment by SamSaffron** [comment]
Daniel closing this in favor of the far simpler core change of simply keeping the number in core and the rest in the plugin.

**Pull request closed without merging**